tests/unittest/_torch/speculative/test_external_api.py

```
@@ -0,0 +1,270 @@
+def setup_server():
+    process = multiprocessing.Process(target=create_server, daemon=True)
+    process.start()
+    # wait for server to start
+    count = 0
+    while count < 10:
+        try:
+            # check if server is running successfully
+            response = httpx.post(f"http://localhost:{PORT}/generate", json={"prefix": [1, 2, 3]})
+            if response.status_code == 200:
+                break
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)